playground/attention_with_rpe_test.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2025 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import math
+import torch
+from torch.profiler import profile, record_function, ProfilerActivity
+from torch.nn import functional as F
+from torch.nn.attention.flex_attention import flex_attention
+from torch.nn.attention.flex_attention import create_block_mask
+from torch.testing import assert_close
+from typing import Any, Callable
+
+from iree.turbine.kernel.gen import TestLaunchContext
+from iree.turbine.kernel.wave.constraints import MMAType
+from iree.turbine.kernel.wave.templates.attention_common import AttentionShape
+from iree.turbine.kernel.wave.utils import (
+    device_randn,
+    device_zeros,
+    get_default_run_config,
+    to_default_device,
+)
+from attention_with_rpe_template import (
+    get_vanilla_attention_kernel as get_vanilla_tkw_attention_with_rpe_output_kernel,
+)
+
+torch.manual_seed(0)
+torch.set_printoptions(
+    linewidth=1000000,
+    threshold=1000000,
+    precision=3,
+)
+
+
+### TKW Harness
+def run(fun: Callable, hparams, *args) -> Any:
+    # with torch.profiler.profile(
+    #     activities=[torch.profiler.ProfilerActivity.CUDA]
+    # ) as prof:
+    with torch.no_grad():  # Disable gradient calculations
+        with TestLaunchContext(
+            hparams,
+            canonicalize=True,
+            # compile_config={"print-ir-after": "all"},
+            run=True,
+            run_config=get_default_run_config(),
+            run_bench=False,
+            schedule=False,
+            use_scheduling_barriers=False,
+        ):
+            fun(*args)
+
+# print(
+#     prof.key_averages(group_by_input_shape=True).table(
+#         sort_by="self_cuda_time_total", row_limit=10
+#     )
+# )
+
+
+#################################################################################
+# INIT VALS
+#################################################################################
+# num_query_heads, num_kv_heads, head_size, head_size_kv
+# WARNING: for now 128^n is the minimal size because attention_with_rpe_template
+# does not support padding and as a consequence we have
```

Comment:
Do you mean padding on the RPE vector? Should we add a mask on RPE ?

Reply:
the vanilla attention part (without RPE) did not agree with torch on sizes 32^n; I suspect this is due to the workgroup / wave constraints that use 128 and 64.
This was what generated those masked stores we talked about earlier; I think (not sure though) this could be a miscompile?